mindsdb/integrations/handlers/youtube_handler/youtube_tables.py

```diff
@@ -96,14 +99,68 @@ def select(self, query: ast.Select) -> pd.DataFrame:
 
         return youtube_comments_df
 
+    def update(self, query: ast.Update) -> pd.DataFrame:
+        """
+        Inserts a new comment in the youtube comments table.
+
+        Args:
+            updated message
+            video_id: The video's unique identifier.
+            comment_id: The comment's unique identifier or ID.
+
+        Returns:
+            Response: Response object containing the results.
+        """
+        update_statement_parser = UPDATEQueryParser(query, self.get_columns())
+        (
+            selected_columns,
+            where_conditions,
+        ) = update_statement_parser.parse_query()
+        params = {}
+        for op, arg1, arg2 in where_conditions:
+            if arg1 == "video_id":
+                if op == "=":
+                    params[arg1] = arg2
+                else:
+                    raise NotImplementedError(f"Invalid operator, {op}. Only the '=' operator is currently supported")
+            if arg1 == "comment_id":
+                if op == "=":
+                    params[arg1] = arg2
+                else:
+                    raise NotImplementedError(f"Invalid operator, {op}. Only the '=' operator is currently supported")
+
+        if "video_id" not in params or "comment_id" not in params:
+            raise Exception(
+                "To update a comment in Youtube, you need to provide the 'video_id' and comment_id parameter."
+            )
+        # Update comment in YouTube video
+        try:
+            body = {"id": params["comment_id"], "snippet": {"textOriginal": selected_columns["comment"]}}
+            update_response = (
+                self.handler.connection.comments()
```

Comment:
You have used `self.connection` throughout your code, however, as far as I know, this should not be accessible prior to calling the `connect()` method of the handler. Can you explain to me how this works?

Reply:
When the handler is first created, the user is redirected to connect to the service. The connection object is then saved as a property of the handler instance, so that the user does not need to go through OAuth verification every time they execute a query.